tailcfg/tailcfg.go

```diff
@@ -1347,9 +1342,12 @@ type MapResponse struct {
 	// ControlTime, if non-zero, is the current timestamp according to the control server.
 	ControlTime *time.Time `json:",omitempty"`
 
-	// TKA, if non-nil, describes updates for the local instance of the
-	// tailnet key authority.
-	TKA *TKAMapResponse `json:",omitempty"`
+	// TKAInfo, if non-nil, describes the control plane's view of tailnet
+	// key authority (TKA) state.
+	//
+	// A nil TKAInfo indicates that the control plane has no information
```

Comment:
now you lost the "nil" means no change bit.

Each MapResponse field has two contexts: the initial response, and deltas. For deltas, they're nowadays all the same: nil means "no change from the previous streamed one". We still document that even though it's the same for all of them. But the initial value also should be documented.

"no information to influence" .... what if the node had NL already enabled and then reconnects and gets nil here? Is that a transition to disabled true?

Reply:
"nil means no change" and "nil means dont touch the local tka" are both true: Because the first time round with the non-nil TKAInfo would have resulted in a local state change. If that makes sense?

Reply:
No, because these fields are all declarative. MapResponse only says the state of the world. The nil-ness is only a compression detail for bandwidth. It has no semantic difference.

Ignoring the compression, if we didn't do that, what does nil mean? It can't mean "don't touch" because it can't mean something imperative. 

Does a nil mean NL isn't enabled?

Reply:
Oh I see what youre trying to do. We could remove `TKAInfo.Enabled` and rely just on nil vs non-nil, and that would be fully declarative, but that has some side-effects we need to accept: A client with a local TKA state talking to a server which doesnt undestand these fields is going to result in an RPC to get the disablement secret (which will 404), because not supporting this field implies NL is disabled.

This is messier than the other fields because NL is fundamentally about not blindly trusting the declarations from control: It always needs to involve consideration of local state.